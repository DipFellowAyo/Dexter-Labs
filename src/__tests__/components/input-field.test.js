import React from 'react'
import TextField from '@material-ui/core/TextField'
import FormControl from '@material-ui/core/FormControl'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

import InputField from '../../components/forms/input-field'

describe('InputField Component', () => {
  let inputField

  beforeEach(() => {
    inputField = shallow(<InputField />)
  })
  afterAll(() => jest.resetAllMocks())

  it('should render without crashing', () => {
    expect(inputField.exists()).toBe(true)
  })

  it('should match the snapshots', () => {
    expect(toJson(inputField)).toMatchSnapshot()
  })

  it('accepts `name` as props', () => {
    expect(inputField.props().name).toBeUndefined()
  })

  it('accepts `onChange` as props', () => {
    expect(inputField.props().onChange).toBeUndefined()
  })

  it('renders a TextField component', () => {
    expect(inputField.find(TextField).exists()).toBe(true)
  })

  it('renders a TextField component with `name` prop', () => {
    expect(inputField.find(TextField).props().name).toBeUndefined()
  })

  it('renders a TextField component with `onChange` prop', () => {
    expect(inputField.find(TextField).props().onChange).toBeUndefined()
  })

  it('renders a FormControl component', () => {
    expect(inputField.find(FormControl).exists()).toBe(true)
  })

  it('renders a FormControl component with `fullWidth` prop', () => {
    expect(inputField.find(FormControl).props().fullWidth).toBe(true)
  })
})
