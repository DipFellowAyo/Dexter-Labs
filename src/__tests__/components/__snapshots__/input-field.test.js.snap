// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Component should match the snapshots 1`] = `
<WithStyles(ForwardRef(FormControl))
  fullWidth={true}
>
  <WithStyles(ForwardRef(TextField)) />
</WithStyles(ForwardRef(FormControl))>
`;
